freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/conversion/UserMetadataConverter.java

```diff
@@ -0,0 +1,100 @@
+package com.sequenceiq.freeipa.service.freeipa.user.conversion;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.google.common.base.Strings;
+import com.google.common.io.BaseEncoding;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.CrnParseException;
+import com.sequenceiq.freeipa.client.model.User;
+import com.sequenceiq.freeipa.service.freeipa.user.model.JsonUserMetadata;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UserMetadata;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMetadataConverter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserMetadataConverter.class);
+
+    private static final ObjectReader OBJECT_READER = new ObjectMapper().readerFor(JsonUserMetadata.class)
```

Comment:
look at `com.sequenceiq.cloudbreak.common.json.JsonUtil` and `Json`. You should be able to use those classes rather than your own OBJECT_READER

Reply:
Do you feel strongly about this? It's not a lot of code to create the reader/writer objects, and it allows this class to control the configuration of the mappers exactly. The fasterxml doc also explicitly says that ObjectReader/Writer are "fully immutable," while ObjectMapper is said to be "thread-safe" as long as it's configured before use. This made me feel better about using ObjectReader/Writer, although I admin I'm being a little paranoid there. There's also a pattern of creating purpose-built object mappers in the CB code base.

Reply:
I'm ok with whatever you decide

Reply:
@handavid, please let me know your thoughts on the above.

Reply:
OK, I'll leave the reader/writer objects in the converter class, then.